modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v10_22_1/ResourceActionUpgradeProcess.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.internal.upgrade.v10_22_1;
+
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.ResourceAction;
+import com.liferay.portal.kernel.model.ResourcePermission;
+import com.liferay.portal.kernel.service.ResourceActionLocalServiceUtil;
+import com.liferay.portal.kernel.service.ResourcePermissionLocalServiceUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Mario Gomes
+ */
+public class ResourceActionUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws PortalException {
+		if (_hasOldResourceAction()) {
+			_upgradeResourcePermission();
+
+			ResourceActionLocalServiceUtil.deleteResourceAction(
+				_resourceActionOld);
+		}
+	}
+
+	private boolean _hasOldResourceAction() {
+		_resourceActionOld = ResourceActionLocalServiceUtil.fetchResourceAction(
+			_OLD_NAME, _OLD_ACTION);
+
+		return Objects.nonNull(_resourceActionOld);
+	}
+
+	private void _upgradeResourcePermission() throws PortalException {
+		List<ResourcePermission> resourcePermissions =
+			ResourcePermissionLocalServiceUtil.getResourcePermissions(
+				_OLD_NAME);
+
+		for (ResourcePermission resourcePermission : resourcePermissions) {
+			long actionIds = resourcePermission.getActionIds();
+
+			for (ResourceAction resourceAction :
+					ResourceActionLocalServiceUtil.getResourceActions(
+						_OLD_NAME)) {
+
+				long bitwiseValue = resourceAction.getBitwiseValue();
+
+				if (((actionIds & bitwiseValue) == bitwiseValue) &&
+					Objects.equals(resourceAction.getActionId(), _OLD_ACTION) &&
+					!Objects.equals(
+						resourcePermission.getPrimKey(), _OLD_NAME)) {
+
+					ResourcePermissionLocalServiceUtil.addResourcePermission(
+						resourcePermission.getCompanyId(), _NEW_NAME,
+						resourcePermission.getScope(),
+						resourcePermission.getPrimKey(),
+						resourcePermission.getRoleId(), _NEW_ACTION);
+				}
+			}
+
+			ResourcePermissionLocalServiceUtil.removeResourcePermission(
+				resourcePermission.getCompanyId(), resourcePermission.getName(),
+				resourcePermission.getScope(), resourcePermission.getPrimKey(),
+				resourcePermission.getRoleId(), _OLD_ACTION);
+		}
+	}
+
+	private static final String _NEW_ACTION = "ADD_OBJECT_ENTRY_FOLDER";
```

Comment:
1 usage and you have the new constant, so use the constant from `ObjectActionKeys`